components/skus/browser/rs/lib/src/sdk/orders.rs

```diff
@@ -238,6 +244,35 @@ where
         Ok(order)
     }
 
+    #[instrument(err(level = Level::WARN), ret)]
+    // submit_receipt_new_order allows for order creation with a vendor receipt, returning the
+    // order id of the created order
+    pub async fn submit_receipt_new_order(&self, receipt: &str) -> Result<String, InternalError> {
+        event!(Level::DEBUG, order_id = order_id, "submit_receipt_new_order called");
+        let request_with_retries = FutureRetry::new(
+            || async {
+                let builder = http::Request::builder()
+                    .method("POST")
+                    .uri(format!("{}/v1/orders/receipt", self.base_url, order_id));
+
+                let receipt_bytes = receipt.as_bytes().to_vec();
+                let req = builder.body(receipt_bytes)?;
+
+                let resp: SubmitReceiptNewOrderResponse = self.fetch(req).await?;
+                match resp.status() {
+                    http::StatusCode::OK => Ok(resp.order_id),
```

Comment:
Also, this should expect a `201` as we discussed, not `200`.

Reply:
addressed: https://github.com/brave/brave-core/pull/21773/commits/8e295eddc53eec6dc0a649158ee3ecd8ba8df20e